src/components/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,54 @@
+import {Link} from 'react-router-dom';
+import { useQuiz } from "../../QuizContext/QuizContext";
+import './Dashboard.css'
+import coverpic1 from '../../Assets/coverpic1.png'
+
+export const Dashboard = () => {
+  const { quizState, quizDispatch } = useQuiz();
+  const takeQuiz = (quizId: string): any => {
+    quizDispatch({
```

Comment:
Update Quiz function can take the responsibility of setting the initial score and question index to avoid multiple dispatch

Reply:
done!